File: Java/TypeInference.java
package demo;

import java.util.ArrayList;

public class TypeInference {
	public static void main(String[] args) {
        ArrayList<String> list1 = new ArrayList<String>();
        ArrayList<String> list2 = new ArrayList<>();

        list1.add("Java");
        list2.add("Kotlin");

        System.out.println(list1);
        System.out.println(list2);
    }
}

File: Kotlin/TypeInference.kt
package demo

fun main() {
    // Khai báo có chỉ rõ kiểu
    val list1: ArrayList<String> = ArrayList<String>()
    
    // Type inference: Kotlin tự suy ra kiểu
    val list2 = ArrayList<String>()

    list1.add("Java")
    list2.add("Kotlin")

    println(list1)
    println(list2)
}
